diesel_cli/src/main.rs

```diff
@@ -194,7 +194,19 @@ fn run_infer_schema(matches: &ArgMatches) {
     let database_url = database::database_url(matches);
     let schema_name = matches.value_of("schema");
 
-    let filtering_tables = matches.values_of("table-name").map(|v| v.collect())
+    let filtering_tables = matches.values_of("table-name").map(|v| {
+        if cfg!(feature = "postgres") {
```

Comment:
Can you move that `if` into the inner `map`? This way, you can combine the `if` there to something like `if !cfg!(feature = "postgres") || v.contains(".")`?